scripts/benchmarks_local.py

```diff
@@ -0,0 +1,450 @@
+    parser.add_argument('--commits', nargs='+', type=str, help='The commits to test.')
+    parser.add_argument('--repo-url', type=str, default='https://github.com/dotnet/runtime.git', help='The runtime repo to test from, used to get data for a fork.')
+    parser.add_argument('--local-test-repo', type=str, help='Path to a local repo with the runtime source code to test from.') 
+    parser.add_argument('--separate-repos', action='store_true', help='Whether to test each runtime version from their own separate repo directory.') # TODO: Do we want to have this as an actual option? It made sense before a shared build cache was added
+    parser.add_argument('--repo-storage-path', type=str, default='.', help='The path to store the cloned repositories in.')
+    parser.add_argument('--artifact-storage-path', type=str, default=f'{os.getcwd()}{os.path.sep}runtime-testing-artifacts', help=f'The path to store the artifacts in (builds, results, etc). Default is {os.getcwd()}{os.path.sep}runtime-testing-artifacts')
+    parser.add_argument('--rebuild-artifacts', action='store_true', help='Whether to rebuild the artifacts for the specified commits before benchmarking.')
+    parser.add_argument('--build-only', action='store_true', help='Whether to only build the artifacts for the specified commits and not run the benchmarks.')
+    parser.add_argument('--skip-local-rebuild', action='store_true', help='Whether to skip rebuilding the local repo and use the already built version (if already built). Useful if you need to run against local changes again.')
+    parser.add_argument('--allow-non-admin-execution', action='store_true', help='Whether to allow non-admin execution of the script. Admin execution is highly recommended as it minimizes the chance of encountering errors, but may not be possible in all cases.')
+    def __is_valid_run_type(value):
+        try:
+            RunType[value]
+        except KeyError:
+            raise ArgumentTypeError(f"Invalid run type: {value}.")
+        return value
+    parser.add_argument('--run-types', dest='run_type_names', nargs='+', type=__is_valid_run_type, choices=[run_type.name for run_type in RunType], help='The types of runs to perform.')
+    parser.add_argument('--quiet', dest='verbose', action='store_false', help='Whether to not print verbose output.')
+    
+    # Arguments specifically for dependency generation and BDN
+    parser.add_argument('--bdn-arguments', type=str, default="", help='Command line arguments to be passed to BenchmarkDotNet, wrapped in quotes')
+    parser.add_argument('--architecture', choices=['x64', 'x86', 'arm64', 'arm'], default=get_machine_architecture(), help='Specifies the SDK processor architecture')
+    parser.add_argument('--os', choices=['windows', 'linux'], default=platform.system().lower(), help='Specifies the operating system of the system')
```

Comment:
It would be good to add support for osx. Currently, it selects darwin which is incorrect os for the runtime.